src/serviceContainer.ts

```diff
@@ -0,0 +1,33 @@
+export class ServiceContainer {
```

Comment:
I think we should enforce singleton pattern on the ServiceContainer, like restricting construction call to 1 time, if you want it to be a global accessor container for keepers.

Reply:
Done.